spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -371,13 +374,20 @@ def get_reviews(app_id, platform, storefront, version_id)
 
         r = request(:get, rating_url)
         all_reviews.concat(parse_response(r, 'data')['reviews'])
+        last_review_date = Time.at(all_reviews[-1]['value']['lastModified']/1000)
+        
+        break if last_review_date < upto_date && !upto_date.nil?
+        
         if all_reviews.count < parse_response(r, 'data')['reviewCount']
           index += per_page
         else
           break
         end
       end
-      all_reviews
+
+      return all_reviews if upto_date.nil?
+
+      return all_reviews.select { |review| Time.at(review['value']['lastModified']/1000) > upto_date  }
```

Comment:
Remove these multiple `return`s and just do... (see above comment)

```rb
return all_reviews
```

Reply:
Done! :)